### PR TITLE
generaldyne pdf

### PR DESCRIPTION
**Pull request by ziofil**
`physics.gaussian.general_dyne()` now returns the pdf for further use (e.g. sampling)

### CODE REVIEW

File: mrmustard/physics/gaussian.py [hunk 1]
```diff
@@ -574,34 +573,35 @@ def general_dyne(
             oucome probability [units of `\hbar**N`],
             post-measurement covariace [units of `2\hbar`]
             post-measurement means vector [units of `\sqrt{\hbar}`].
+            pdf (Tensorflow probability distribution)
```
Comment (SamFerracin) [inline mrmustard/physics/gaussian.py:576]: If the numpy backend is used, it is not a tensorflow distribution

Comment (ziofil) [reply mrmustard/physics/gaussian.py:576]: Yes it is. Here we are using TF Probability, not TF.

**Comment by ziofil** [comment]
> This PR adds "pdf" as an output for the generaldyne measurement and fixes small bugs.
> 
> Just have a small question for @ziofil before approving it, I haven't found any corresponding tests in tests_gaussian for this, do we need add one?

I'm not sure what you want to test here

**Review by sylviemonet — APPROVED** [review]

File: mrmustard/physics/gaussian.py (hunk 1 above)
Comment (SamFerracin) [reply mrmustard/physics/gaussian.py:576]: No, I don't think so. Look at the MultivariateNormalTriL function in numpy_backend.py. It does not return a TF object. Let's touch base on this later, before you resolve this convo

**Pull request closed without merging**